include/cli/modules/deploy.php

```diff
@@ -277,8 +276,12 @@ function run($args, $options) {
                 "*/.htaccess", ".MANIFEST"));
         }
 
-        if (!$options['dry-run'])
-            $this->writeManifest($this->destination);
+        if (!$options['dry-run']) {
+            $this->writeManifest();
+
+            $this->clean($root, $this->destination, -1, array($include, "setup/"));
+            $this->clean("$root/include", $include, -1, array("ost-config.php","settings.php","plugins/", "*/.htaccess"));
```

Comment:
There's an option for cleaning. IMO, since it could delete things an administrator might want to keep, it should be run explicitly with the option, perhaps with the `dry-run` flag as well, before permanently removing things